src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorTask.cs

```diff
@@ -0,0 +1,505 @@
+            {
+                _logger.LogInformation(canceledEx, "Import task canceled. {0}", canceledEx.Message);
+
+                await CancelProcessingTasksAsync();
+                taskResultData = new TaskResultData(TaskResult.Canceled, canceledEx.Message);
+            }
+            catch (IntegrationDataStoreException integrationDataStoreEx)
+            {
+                _logger.LogInformation(integrationDataStoreEx, "Failed to access input files.");
+
+                ImportTaskErrorResult errorResult = new ImportTaskErrorResult();
+                errorResult.HttpStatusCode = integrationDataStoreEx.StatusCode;
+                errorResult.ErrorMessage = integrationDataStoreEx.Message;
+
+                taskResultData = new TaskResultData(TaskResult.Fail, JsonConvert.SerializeObject(errorResult));
+            }
+            catch (ImportFileEtagNotMatchException eTagEx)
+            {
+                _logger.LogInformation(eTagEx, "Import file etag not match.");
+
+                ImportTaskErrorResult errorResult = new ImportTaskErrorResult();
+                errorResult.HttpStatusCode = HttpStatusCode.BadRequest;
+                errorResult.ErrorMessage = eTagEx.Message;
+
+                taskResultData = new TaskResultData(TaskResult.Fail, JsonConvert.SerializeObject(errorResult));
+            }
+            catch (ImportProcessingException processingEx)
+            {
+                _logger.LogInformation(processingEx, "Failed to process input resources.");
+
+                ImportTaskErrorResult errorResult = new ImportTaskErrorResult();
+                errorResult.HttpStatusCode = HttpStatusCode.BadRequest;
+                errorResult.ErrorMessage = processingEx.Message;
+
+                taskResultData = new TaskResultData(TaskResult.Fail, JsonConvert.SerializeObject(errorResult));
+            }
+            catch (Exception ex)
+            {
+                _logger.LogInformation(ex, "Failed to import data.");
+
+                ImportTaskErrorResult errorResult = new ImportTaskErrorResult();
+                errorResult.HttpStatusCode = HttpStatusCode.InternalServerError;
+                errorResult.ErrorMessage = ex.Message;
+
+                throw new RetriableTaskException(JsonConvert.SerializeObject(errorResult));
+            }
+
+            if (_orchestratorTaskContext.Progress > ImportOrchestratorTaskProgress.InputResourcesValidated)
+            {
+                // Post-process operation cannot be cancelled.
+                try
+                {
+                    await _fhirDataBulkImportOperation.DeleteDuplicatedResourcesAsync(CancellationToken.None);
+                    await _fhirDataBulkImportOperation.PostprocessAsync(CancellationToken.None);
+
+                    _logger.LogInformation("Postprocess Completed");
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogInformation(ex, "Failed at postprocess step.");
```

Comment:
Should something be done with taskResultData in the event that one of the earlier steps hit an exception and then the post process steps hit an exception? Currently it looks like there is no way to tell if two exceptions were hit.

Reply:
Put inner error for multiple error scenario.